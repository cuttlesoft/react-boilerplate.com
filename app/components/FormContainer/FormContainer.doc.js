import { describe, PropTypes } from 'react-desc'

export const doc = FormContainer => {
  const DocumentedFormContainer = describe(FormContainer)
    .description('A FormContainer')
    .usage(
      `import { FormContainer } from 'components/FormContainer';
<FormContainer />`,
    )

  DocumentedFormContainer.propTypes = {
    id: PropTypes.string
      .description('The DOM id attribute value to use for the element.')
      .defaultValue(false),
  }

  return DocumentedFormContainer
}
