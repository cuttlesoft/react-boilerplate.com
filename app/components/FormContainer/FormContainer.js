import React from 'react'
import PropTypes from 'prop-types'

// Components
import { Box } from '../Box'

/**
 * FormContainer
 */
const FormContainer = ({ children }) => (
  <Box margin="auto" pad="large" style={{ width: '90vw', maxWidth: 600 }}>
    {children}
  </Box>
)

FormContainer.propTypes = {
  children: PropTypes.any,
}

export default FormContainer
