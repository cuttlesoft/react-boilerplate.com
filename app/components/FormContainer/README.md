<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## FormContainer
A FormContainer

## Usage

```javascript
import { FormContainer } from 'components/FormContainer';
<FormContainer />
```

## Properties

**id**

The DOM id attribute value to use for the element.

```
string
```
  