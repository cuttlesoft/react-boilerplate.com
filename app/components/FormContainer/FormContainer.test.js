import React from 'react'
import { render } from '@testing-library/react'
// import ''@testing-library/jest-dom/extend-expect' // add some helpful assertions

import FormContainer from './FormContainer'

const renderComponent = (props = {}) => render(<FormContainer {...props} />)

/**
 *
 * Tests for FormContainer
 *
 */
describe('FormContainer', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('has additional unit tests specified', () => {
    expect(true).toEqual(false)
  })
})
