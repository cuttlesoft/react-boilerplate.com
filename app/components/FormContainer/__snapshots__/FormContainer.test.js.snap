// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContainer renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 ewTzDI"
    style="width: 90vw; max-width: 600px;"
  />
</div>
`;
